core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -61,6 +61,21 @@ class StageInfo(
       "running"
     }
   }
+
+  private[spark] def getStatusDetail: String = {
```

Comment:
Is this used elsewhere or can it just go into `SparkListener`?

I think this is confusing as one big format string. I'd make `val`s for the pieces that need some computation and return one big interpolated string. Also this is verbose:

```
foo match {
  Some(x) => f(x)
  case _ => ""
}
```
compared to just `foo.map(x => f(x)).getOrElse("")`